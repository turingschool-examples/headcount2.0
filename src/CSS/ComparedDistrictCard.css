@font-face {
  font-family: anon;
  src: url(../Fonts/AnonymousPro-Regular.ttf);
}

.compared-district-card0 {
  animation: shazam 1s linear, bazinga 1s ease-in;
  animation: shazam 1s linear, bazinga 1s ease-out;
  grid-column: 1 / 2;
  justify-self: center;
  align-self: center;
}

.compared-district-card1 {
  animation: shazam 1s linear, bazinga 1s ease-in;
  animation: shazam 1s linear, bazinga 1s ease-out;
  grid-column: 3 / 4;
  justify-self: center;
  align-self: center;
}

.compared-location {
  text-align: center;
  font-size: 1.5rem;
}

.year-red {
  font-family: anon;
  text-align: center;
  color: rgb(169, 77, 77);
}

.year-black {
  font-family: anon;
  text-align: center;
  color: green;
}

@keyframes shazam {
  0% {
    right: 250px;
  }
  100% {
    right: 10px;
  }
}

@keyframes bazinga {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
