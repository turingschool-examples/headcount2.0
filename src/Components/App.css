html {
  background-color: #2a393a;
  color: #f4fcfc;
  font-size: 62.5%;
}

h1 {
  color: #f7b488;
  font-size: 3.5em;
  margin: 0 0 1em 0;
  padding: .5em;
  text-align: center;
}