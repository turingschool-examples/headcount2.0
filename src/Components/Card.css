.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}