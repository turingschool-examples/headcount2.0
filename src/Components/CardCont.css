.card-cont {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}