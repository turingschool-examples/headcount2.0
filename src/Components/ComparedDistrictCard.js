import React from "react";
import "../CSS/ComparedDistrictCard.css";
import PropTypes from "prop-types";

export const ComparedDistrictCard = ({ location, stats, index }) => {
  return (
    <div className={"compared-district-card" + index}>
      <h1 className="compared-location">{location}</h1>
      {Object.keys(stats).map((year, index) => {
        const avg = stats[year];
        return (
          <section
            key={`avgStats ${index}`}
            className={avg < 0.5 ? "year-red" : "year-black"}
          >
            {year}
            <span className={avg < 0.5 ? "year-red" : "year-black"}>
              ....
              {avg}
            </span>
          </section>
        );
      })}
      <p />
    </div>
  );
};

const { string, func, number, shape } = PropTypes;

ComparedDistrictCard.propTypes = {
  location: string,
  stats: shape({ year: number }),
  selectCard: func,
  index: number
};
