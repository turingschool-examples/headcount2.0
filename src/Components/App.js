import React, { Component } from "react";
import DistrictRepository from "../helper";
import kindergarners from "../data/kindergartners_in_full_day_program";
import Search from "./Search";
import { DistrictCardContainer } from "./DistrictCardContainer";
import { ControlCards } from "./ControlCards";
import "../CSS/App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      districts: [],
      selectedDistricts: [],
      comparisonData: {}
    };
    this.districtRepository = new DistrictRepository(kindergarners);
  }

  componentDidMount = () => {
    const districts = this.districtRepository.districtsArray();
    this.setState({ districts });
  };

  handleSubmit = district => {
    this.setState({
      districts: district
    });
  };

  selectCard = location => {
    const { districts } = this.state;
    let targetDistrict = districts.find(
      district => district.location === location
    );
    const selectedDistricts = this.checkFoundDistrict(targetDistrict);
    this.setState({ selectedDistricts, districts });
    if (selectedDistricts.length === 2) {
      this.compareDistricts(
        selectedDistricts[0].location,
        selectedDistricts[1].location
      );
    }
  };

  checkFoundDistrict = district => {
    let targetIndex = this.state.districts.indexOf(district);
    let selectedDistricts;
    if (!district["clicked"] && this.state.selectedDistricts.length < 2) {
      district["clicked"] = true;
      selectedDistricts = [...this.state.selectedDistricts, district];
      this.setState({ districts: [this.state.districts[targetIndex]] });
    } else {
      district["clicked"] = false;
      selectedDistricts = this.state.selectedDistricts.filter(
        district => district.clicked === true
      );
    }
    return selectedDistricts;
  };

  clearComparisons = () => {
    this.setState({ comparisonData: {} });
  };

  compareDistricts = (districtOne, districtTwo) => {
    const districtOneInfo = this.districtRepository.findByName(districtOne);
    const districtTwoInfo = this.districtRepository.findByName(districtTwo);
    const districtComparison = this.districtRepository.compareDistrictAverages(
      districtOne,
      districtTwo
    );
    const comparisonData = Object.assign({}, { ...districtComparison });
    comparisonData["comparisonData"] = {
      [districtOne]: comparisonData[districtOne],
      [districtTwo]: comparisonData[districtTwo],
      compared: comparisonData.compared
    };
    delete comparisonData.compared;
    comparisonData[districtOne] = districtOneInfo;
    comparisonData[districtTwo] = districtTwoInfo;
    this.setState({ comparisonData });
  };

  render() {
    return (
      <div className="head-wrapper">
        <Search
          selectedDistricts={this.state.selectedDistricts}
          selectCard={this.selectCard}
          handleSubmit={this.handleSubmit}
          clearComparisons={this.clearComparisons}
        />
        <div
          className={
            this.state.selectedDistricts.length > 1
              ? "title-hide"
              : "title-show"
          }
        >
          HeadCount 2.0
          <p className="instructions">
            Click Districts below for Annual Score Avg.
          </p>
        </div>
        {this.state.comparisonData !== {} && (
          <ControlCards
            selectedDistricts={this.state.selectedDistricts}
            comparisonData={this.state.comparisonData}
            selectCard={this.selectCard}
          />
        )}
        {this.state.districts && (
          <DistrictCardContainer
            districts={this.state.districts}
            selectCard={this.selectCard}
          />
        )}
      </div>
    );
  }
}

export default App;
