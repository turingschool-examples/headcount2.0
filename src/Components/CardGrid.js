/* eslint-disable */
import React from 'react'
import Card from './Card.js'
import DistrictRepository from '../Helpers/helper'
import kinderData from '../../data/kindergartners_in_full_day_program.js';

const CardGrid =({ schools, searched, cardClick, cardUnclick }) => {

  return(
    <section className="card-grid">
      {schools.findAllMatches(searched).map((school, i) =>{
        return(
          <div key = {i} >
             <Card location = {school.location}
                   data = {school.data}
                   schools = {schools}
                   cardClick = {cardClick}
                   cardUnclick = {cardUnclick}
                   id = {i}/>
          </div>
        )
      })}
    </section>
  )
}

export default CardGrid
