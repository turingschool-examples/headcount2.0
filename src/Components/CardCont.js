import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './CardCont.css';

const CardCont = ({ data, selectCard, comparedData, compareSelectedCards }) => {
  const displayDistrictCards = Object.values(data).map((district, i) => {
    return  <Card location={ district.location }
                  stats={ district.stats }
                  isSelected={ district.isSelected }
                  key={ `card-${i}` }
                  selectCard={selectCard}
            />
  })

  return (
    <article className="card-cont">
      { displayDistrictCards }
    </article>
  )
}

CardCont.propTypes = {
  data: PropTypes.array
}

export default CardCont;