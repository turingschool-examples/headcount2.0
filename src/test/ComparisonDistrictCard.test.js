import React from "react";
import { ComparedDistrictCard } from "../Components/ComparedDistrictCard";

describe("ComparedDistrictCard", () => {
  let shallowWrap;
  let mockStats = {
    2004: 0.228,
    2005: 0.3,
    2006: 0.293,
    2007: 0.306,
    2008: 0.673,
    2009: 1,
    2010: 1,
    2011: 1,
    2012: 1,
    2013: 0.998,
    2014: 1
  };
  beforeEach(() => {
    shallowWrap = shallow(
      <ComparedDistrictCard
        location={"ACADEMY 20"}
        stats={mockStats}
        index={1}
      />
    );
  });

  it("should exist", () => {
    expect(shallowWrap).toBeDefined();
  });

  it("should match snapshot", () => {
    shallowWrap = shallow(
      <ComparedDistrictCard location={""} stats={{}} index={null} />
    );
    expect(shallowWrap).toMatchSnapshot();
  });

  it("should match snapshot", () => {
    expect(shallowWrap).toMatchSnapshot();
  });
});
