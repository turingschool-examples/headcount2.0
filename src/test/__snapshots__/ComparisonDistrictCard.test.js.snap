// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparedDistrictCard should match snapshot 1`] = `
<div
  className="compared-district-cardnull"
>
  <h1
    className="compared-location"
  />
  <p />
</div>
`;

exports[`ComparedDistrictCard should match snapshot 2`] = `
<div
  className="compared-district-card1"
>
  <h1
    className="compared-location"
  >
    ACADEMY 20
  </h1>
  <section
    className="year-red"
    key="avgStats 0"
  >
    2004
    <span
      className="year-red"
    >
      ....
      0.228
    </span>
  </section>
  <section
    className="year-red"
    key="avgStats 1"
  >
    2005
    <span
      className="year-red"
    >
      ....
      0.3
    </span>
  </section>
  <section
    className="year-red"
    key="avgStats 2"
  >
    2006
    <span
      className="year-red"
    >
      ....
      0.293
    </span>
  </section>
  <section
    className="year-red"
    key="avgStats 3"
  >
    2007
    <span
      className="year-red"
    >
      ....
      0.306
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 4"
  >
    2008
    <span
      className="year-black"
    >
      ....
      0.673
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 5"
  >
    2009
    <span
      className="year-black"
    >
      ....
      1
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 6"
  >
    2010
    <span
      className="year-black"
    >
      ....
      1
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 7"
  >
    2011
    <span
      className="year-black"
    >
      ....
      1
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 8"
  >
    2012
    <span
      className="year-black"
    >
      ....
      1
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 9"
  >
    2013
    <span
      className="year-black"
    >
      ....
      0.998
    </span>
  </section>
  <section
    className="year-black"
    key="avgStats 10"
  >
    2014
    <span
      className="year-black"
    >
      ....
      1
    </span>
  </section>
  <p />
</div>
`;
